components/patient-data/impl/src/test/java/org/phenotips/data/internal/controller/APGARControllerTest.java

```diff
@@ -0,0 +1,259 @@
+import org.xwiki.bridge.DocumentAccessBridge;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.test.mockito.MockitoComponentMockingRule;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.Map;
+
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Tests for the {@link APGARController} Component,
+ * implementation of the {@link org.phenotips.data.PatientDataController} interface
+ */
+public class APGARControllerTest
+{
+    @Rule
+    public MockitoComponentMockingRule<PatientDataController> mocker =
+            new MockitoComponentMockingRule<PatientDataController>(APGARController.class);
+
+    private static final String DATA_NAME = "apgar";
+
+    private static final String APGAR_1 = "apgar1";
+
+    private static final String APGAR_5 = "apgar5";
+
+    private Logger logger;
+
+    private DocumentAccessBridge documentAccessBridge;
+
+    private APGARController controller;
+
+    @Mock
+    private Patient patient;
+
+    @Mock
+    private XWikiDocument doc;
+
+    @Mock
+    private BaseObject data;
+
+    @Before
+    public void setUp() throws Exception
+    {
+        MockitoAnnotations.initMocks(this);
+
+        this.controller = (APGARController)this.mocker.getComponentUnderTest();
+        this.logger = this.mocker.getMockedLogger();
+        this.documentAccessBridge = this.mocker.getInstance(DocumentAccessBridge.class);
+
+        DocumentReference patientDocument = new DocumentReference("wiki", "patient", "00000001");
+        doReturn(patientDocument).when(this.patient).getDocument();
+        doReturn(this.doc).when(this.documentAccessBridge).getDocument(patientDocument);
+        doReturn(this.data).when(this.doc).getXObject(Patient.CLASS_REFERENCE);
+    }
+
+    @Test
+    public void loadCatchesExceptionWhenPatientDoesNotHavePatientClass()
+    {
+        doReturn(null).when(this.doc).getXObject(Patient.CLASS_REFERENCE);
+
+        PatientData<Integer> result = this.controller.load(this.patient);
+
+        verify(this.logger).error("Could not find requested document or some unforeseen"
+                        + " error has occurred during controller loading ",
+                PatientDataController.ERROR_MESSAGE_NO_PATIENT_CLASS);
+        Assert.assertNull(result);
+    }
+
+    @Test
+    public void loadDoesNotReturnNullIntegers()
+    {
+        doReturn(null).when(this.data).getStringValue(anyString());
+
+        PatientData<Integer> result = this.controller.load(this.patient);
+
+        Assert.assertEquals(0, result.size());
+    }
+
+    @Test
+    public void loadDoesNotReturnNonIntegerStrings()
+    {
+        doReturn("STRING").when(this.data).getStringValue(anyString());
+
+        PatientData<Integer> result = this.controller.load(this.patient);
+
+        Assert.assertEquals(0, result.size());
```

Comment:
Instead of this, `Assert.assertTrue(result.isEmpty())` might make what you're testing more clear.